include/system/dev/vfs.h

```diff
@@ -28,6 +28,14 @@ struct fs_driver {
 	int (*isatty_r)(struct _reent*, void* const);
 	off_t (*lseek_r)(struct _reent*, void* const, off_t, int);
 	int (*ctl)(void* const, const uint32_t, void* const);
+	int (*mkdir_r)(struct _reent*, void* const, mode_t mode);
```

Comment:
Nitpick: Why does mode_t have it's argument name, and nothing else does? I'd remove the argument name.